### PR TITLE
Add search bar to manage suppliers page

### PR DESCRIPTION
**Pull request by bjss-colin**

### CODE REVIEW

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Admin/Controllers/SuppliersController.cs [hunk 1]
```diff
@@ -24,15 +25,35 @@ public SuppliersController(ISuppliersService suppliersService)
...
+        {
+            var currentPageUrl = new UriBuilder(HttpContext.Request.Headers.Referer.ToString());
+            var suppliers = await suppliersService.GetSuppliersBySearchTerm(search);
+
+            return Json(suppliers.Select(r =>
+                new AutocompleteResult
+                {
+                    Title = r.Name,
+                    Category = r.Id.ToString(),
+                    Url = currentPageUrl.AppendQueryParameterToUrl(nameof(search), r.Id.ToString()).ToString(),
```
Comment (leerogers-bjss) [inline src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Admin/Controllers/SuppliersController.cs:53]: on the supplier page, the URL should be a direct link to the suppliers edit page. the "CurrentPage" version is only for the marketing page.

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/wwwroot/ts/InactiveItemManagement.ts [hunk 2]
```diff
@@ -1,14 +1,14 @@
 ﻿window.onload = function () {
     const inactiveItemsCheckboxContainer = "inactive-items-checkbox";
     const inactiveItemsCheckbox = "ShowInactiveItems";
+    const checkbox = <HTMLInputElement>document.getElementById(inactiveItemsCheckbox);
```
Comment (leerogers-bjss) [inline src/NHSD.GPIT.BuyingCatalogue.WebApp/wwwroot/ts/InactiveItemManagement.ts:4]: the concept should be not to show the checkbox if you're searching. to facilitate this, magnus did a change that if there is a search term, it wouldn't fire this JS atall, so will need to talk to you/him about it

**Review by leerogers-bjss — CHANGES_REQUESTED** [review]

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Admin/Controllers/SuppliersController.cs (hunk 1 above)
Comment (bjss-colin) [reply src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Admin/Controllers/SuppliersController.cs]: Updated to link to the "Edit Supplier" page

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/wwwroot/ts/InactiveItemManagement.ts (hunk 2 above)
Comment (bjss-colin) [reply src/NHSD.GPIT.BuyingCatalogue.WebApp/wwwroot/ts/InactiveItemManagement.ts]: I've plagiarized Magnus' change to replicate this behavior. Although I have slightly tweaked it as per the acceptance criteria so it disables the inactive filter, on the results page, irrespective as to whether there are any inactive suppliers.

**Review by leerogers-bjss — APPROVED** [review]
technical approval

**Review by Okechukwu802 — APPROVED** [review]

**Review by leerogers-bjss — APPROVED** [review]
technical approval

**Review by leerogers-bjss — APPROVED** [review]
technical approval

**Pull request merged**